iphone/Classes/GeolocationModule.m

```diff
@@ -955,7 +956,11 @@ -(void)setPurpose:(NSString *)reason
 
 #pragma mark Geolacation Analytics
 
--(void)fireApplicationAnalyticsIfNeeded:(NSArray *)locations{
+-(void)fireApplicationAnalyticsIfNeeded:(NSArray *)locations
+{
+    if ([AnalyticsModule isEventFltered:@"ti.geo"]) {
+        return;
+    }
     static BOOL analyticsSend = NO;
```

Comment:
Was trying to test and realised ti.geo is never fired anyway, both with and without the new code. have to remove 'static' here, otherwise analyticsSend will always be set to YES.